### ISSUE TITLE
multisend implementation always need mpi 

### ISSUE THREAD

**Issue by pramodk** [BlueBrain/CoreNeuron#46]
We need ability to build coreneuron without mpi. It seems like multi-send implementation broke this compatibility:

```
[ 17%] Building C object coreneuron/CMakeFiles/coreneuron.dir/scopmath_core/abort.c.o
cd /gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/spack-build/coreneuron && /gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/lib/spack/env/gcc/gcc  -DCORENEURON_BUILD -DDISABLE_HOC_EXP -DENABLE_REPORTING -DHAVE_MALLOC_H -DLAYOUT=0 -DNRNMPI=0 -DNRN_MULTISEND=0 -I/gpfs/bbp.cscs.ch/scratch/gss/viz/kumbhar/neuron_music_home/install/linux-rhel6-x86_64/gcc-4.9.0/reportinglib-develop-aiij3d/include -I/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/utils/randoms -I/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/spack-build/coreneuron -I/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron -I/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron  -g -O2 -Wno-unknown-pragmas   -o CMakeFiles/coreneuron.dir/scopmath_core/abort.c.o   -c /gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/scopmath_core/abort.c
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:89:5: error: ‘NRNMPI_Spike’ does not name a type
     NRNMPI_Spike** buffer_;
     ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp: In function ‘void nrn_multisend_send(PreSyn*, double, NrnThread*)’:
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:116:17: error: ‘class PreSyn’ has no member named ‘multisend_index_’
     int i = ps->multisend_index_;
                 ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:124:9: error: ‘NRNMPI_Spike’ was not declared in this scope
         NRNMPI_Spike spk;
         ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:125:9: error: ‘spk’ was not declared in this scope
         spk.gid = ps->output_index_;
         ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:133:53: error: ‘nrnmpi_multisend’ was not declared in this scope
             nrnmpi_multisend(&spk, cnt_phase1, ranks);
                                                     ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp: In function ‘void multisend_send_phase2(InputPreSyn*, int, double)’:
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:141:17: error: ‘class InputPreSyn’ has no member named ‘multisend_phase2_index_’
     int i = ps->multisend_phase2_index_;
                 ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:147:5: error: ‘NRNMPI_Spike’ was not declared in this scope
     NRNMPI_Spike spk;
     ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:148:5: error: ‘spk’ was not declared in this scope
     spk.gid = gid;
     ^
/gpfs/bbp.cscs.ch/home/kumbhar/workarena/systems/lugviz/softwares/sources/spack/var/spack/stage/coreneuron-develop-ei3gfd37guvrzwgiw7n3hkj5zaitpx5e/CoreNeuron/coreneuron/nrniv/multisend.cpp:150:45: error: ‘nrnmpi_multisend’ was not declared in this scope
     nrnmpi_multisend(&spk, cnt_phase2, ranks);
```

**Comment by nrnhines** [comment]
Hi Pramod.
Since there is no transfer of spikes with one process, Would it be
convenient to surround the entire multisend.cpp and multisend_setup.cpp
files with a
#if NRN_MULTISEND
#endif
I see that NRNMPI and NRN_MULTISEND are set by cmake  to 0
I presume that if NRNMPI is 0 then cmake can force NRN_MULTISEND to 0
-Michael

**Comment by pramodk** [comment]
Ok, I did suggested fix in #47.
